Classes/Domain/Repository/ImageGalleryRepository.php

```diff
@@ -93,69 +94,55 @@ public function findByDemand(array|Matcher $demand = [], array $orderings = [],
             ->leftJoin(
-                'sys_category_record_mm0',
+                'mm',
                 'sys_category',
-                'sys_category0',
-                $queryBuilder->expr()->andX(
-                    $queryBuilder->expr()->eq('sys_category_record_mm0.uid_local', 'sys_category0.uid'),
-                    $queryBuilder->expr()->eq('sys_category0.deleted', $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)),
-                    $queryBuilder->expr()->lte('sys_category0.t3ver_state', $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)),
-                    $queryBuilder->expr()->eq('sys_category0.t3ver_wsid', $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)),
-                    $queryBuilder->expr()->orX(
-                        $queryBuilder->expr()->eq('sys_category0.t3ver_oid', $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)),
-                        $queryBuilder->expr()->eq('sys_category0.t3ver_state', $queryBuilder->createNamedParameter(4, \PDO::PARAM_INT))
-                    ),
-                    $queryBuilder->expr()->eq('sys_category0.hidden', $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)),
-                    $queryBuilder->expr()->lte('sys_category0.starttime', $queryBuilder->createNamedParameter($timestamp, \PDO::PARAM_INT)),
-                    $queryBuilder->expr()->orX(
-                        $queryBuilder->expr()->eq('sys_category0.endtime', $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)),
-                        $queryBuilder->expr()->gt('sys_category0.endtime', $queryBuilder->createNamedParameter($timestamp, \PDO::PARAM_INT))
-                    ),
-                    $queryBuilder->expr()->in('sys_category0.sys_language_uid', [0, -1])
-                )
+                'category',
+                'mm.uid_local = category.uid'
             )
             ->where(
-                $queryBuilder->expr()->andX(
-                    $queryBuilder->expr()->eq('sys_file.type', $queryBuilder->createNamedParameter(2, \PDO::PARAM_INT)),
-                    $queryBuilder->expr()->in('sys_file.uid', (array)$uids)
-                )
+                $queryBuilder->expr()->eq('sys_file.type', 2),
+                $queryBuilder->expr()->in('sys_file.uid', (array)$uids),
+                // Metadata conditions
+                $queryBuilder->expr()->lte('metadata.t3ver_state', 0),
+                $queryBuilder->expr()->eq('metadata.t3ver_wsid', 0),
+                $queryBuilder->expr()->in('metadata.sys_language_uid', [0, -1]),
+                // Category conditions
+                $queryBuilder->expr()->eq('category.deleted', 0),
+                $queryBuilder->expr()->lte('category.t3ver_state', 0),
+                $queryBuilder->expr()->eq('category.t3ver_wsid', 0),
+                $queryBuilder->expr()->eq('category.hidden', 0),
+                $queryBuilder->expr()->lte('category.starttime', $timestamp),
+                $queryBuilder->expr()->or(
```

Comment:
@compaoreh338 We can remove the so-called enabledFields. That's the job of TYPO3 CMS